Healthy/Classes/Extensions/UIButton+Style.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  UIButton+Style.swift
+//  Healthy
+//
+//  Created by Ahmed Nasr on 13/05/2023.
+//
+
+import UIKit
+
+
+//MARK: - button style -
+//
+extension UIButton {
+    enum buttonStyle {
+        case primaryButton
+        case secondaryButton
+        case disableButton
+    }
+}
+
+
+//MARK: - apply button style -
+//
+extension UIButton {
+    func applyButtonStyle(style: buttonStyle) {
+        switch style {
+        case .primaryButton:
+            primaryButtonStyle()
+        case .secondaryButton:
+            secondaryButtonStyle()
+        case .disableButton:
+            disableButtonStyle()
+        }
+    }
+}
+
+
+//MARK: - button style attribute -
+//
+extension UIButton {
+    private func commonButtonStyle() {
```

Comment:
The `commonButtonStyle` method is called in all cases, why don't we create a function that have all the constant repeated logic and a parameter for the style?
 The final result would be something like 
``` swift
extension UIButton {
  enum ButtonStyle {
    case primary
    case secondary
    case disable
  }
}

extension UIButton {
  func HLButton(style: ButtonStyle) {
    switch style {
    case .primary:
      backgroundColor = .black
    case .secondary:
      backgroundColor = .green
      
    case .disable:
      backgroundColor = .gray
    }
    tintColor = .white
    titleLabel?.font = UIFont.systemFont(ofSize: 32)
    layer.cornerRadius = 12
    layer.masksToBounds = true
    NSLayoutConstraint.activate([
      heightAnchor.constraint(equalToConstant: 40),
    ])
  }
}
//  incrementButton.HLButton(style: .secondary)
```

Reply:
Great enhance ♥️